src/main/java/com/code_intelligence/jazzer/mutation/mutator/lang/ByteArrayMutatorFactory.java

```diff
@@ -130,7 +133,94 @@ public byte[] mutate(byte[] value, PseudoRandom prng) {
 
     @Override
     public byte[] crossOver(byte[] value, byte[] otherValue, PseudoRandom prng) {
-      return value;
+      // Passed in values are expected to already honor the min/max length constraints.
+      // As there does not seem to be an easy way to call libFuzzer's internal cross over
+      // algorithm, it is re-implemented in native Java. The algorithm is based on:
+      // https://github.com/llvm/llvm-project/blob/main/compiler-rt/lib/fuzzer/FuzzerMutate.cpp#L440
+      // https://github.com/llvm/llvm-project/blob/main/compiler-rt/lib/fuzzer/FuzzerCrossOver.cpp#L19
+      //
+      // TODO: Measure if this is fast enough.
+
+      byte[] out = null;
+      while (out == null) {
+        switch (prng.indexIn(3)) {
+          case 0:
+            out = intersect(value, otherValue, prng);
+            break;
+          case 1:
+            out = insertPart(value, otherValue, prng);
+            break;
+          case 2:
+            out = overwritePart(value, otherValue, prng);
+            break;
+          default:
+            throw new AssertionError("Invalid cross over function.");
+        }
+      }
+      return enforceLength(out);
+    }
+
+    private static byte[] intersect(byte[] value, byte[] otherValue, PseudoRandom prng) {
+      int maxOutSize = prng.closedRange(0, Math.min(value.length, otherValue.length));
+      byte[] out = new byte[maxOutSize];
+      int outPos = 0;
+      int valuePos = 0;
+      int otherValuePos = 0;
+      boolean usingFirstValue = true;
+      while (
+          outPos < out.length && (valuePos < value.length || otherValuePos < otherValue.length)) {
```

Comment:
I wonder if the expression after `&&` can ever be false while the part before `&&` is true. In that case some of the bytes in `out` would never be initialized.

Reply:
That should be pretty much the libFuzzer cross over algorithm. But you're right. As `out` is at most as long as the smaller of the arrays this can not happen and it's also checked further down. Removed the check in the `while` condition. 

That being said, WDYT about picking a different size for `out`? Always the bigger one?

Reply:
Or it could be between 0 and max(value.length, otherValue.length). So the cross over partially mixes both, up to whatever length. Then the checks can stay. WDYT? :smile:

Reply:
I'm hesitant with that, as it would create bigger arrays and possibly run into OOMs faster.